Lib/Protocols/IdHL7.pas

```diff
@@ -278,17 +293,18 @@   TIdHL7 = class(TIdBaseComponent)
     procedure StartServer;
     procedure StopServer;
     procedure DropServerConnection;
-    procedure ServerConnect(AContext: TIdContext);
-    procedure ServerExecute(AContext: TIdContext);
-    procedure ServerDisconnect(AContext: TIdContext);
+    Procedure ServerConnect(AThread: TIdPeerThread);
+    Procedure ServerExecute(AThread: TIdPeerThread);
+    Procedure ServerDisconnect(AThread: TIdPeerThread);
 
     procedure CheckClientParameters;
     procedure StartClient;
     procedure StopClient;
     procedure DropClientConnection;
+    Procedure ReConnectFromTimeout;
 
-    procedure HandleIncoming(const AMsg : String; AConnection: TIdTCPConnection);
-    function HandleMessage(const AMsg: String; AConn: TIdTCPConnection; var VReply: String): Boolean;
+    Procedure HandleIncoming(Var VBuffer: AnsiString; AConnection: TIdTCPConnection);

```

Comment:
Why change the AMsg parameter from const to var?

Reply:
Based on original authors code (new 2013 one), he seems to expect multiple message blocks in the buffer not just he one (Extracts the one block out of the buffer as needed, thus the Var ). Honestly we can look to change this logic maybe in future if there might be a reason for it as I cannot think of a scenario currently where he will get multiple blocks like this except if somehow he wants to allow bulk hl7 upload via one connection. In short it is as the original author had it and I kept it as is. hope it helps